// @volkswagen-onehub/components-core/dist/container/demo.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const breakpoint_wrapper_1 = require("../breakpoint-wrapper");
const index_1 = require("../demo/demo-root/index");
const helpers_1 = require("../helpers");
const theme_provider_1 = require("../theme-provider");
const index_2 = require("./index");
const demo_1 = require("../demo");
exports.DemoContent = theme_provider_1.styled.div.withConfig({ displayName: "DemoContent", componentId: "sc-tc7e8z" }) `
	color: ${props => props.theme.text.color.primary};
	border: 1px solid ${props => props.theme.text.color.primary};
	padding: 5px;
`;
exports.LargeDemoContent = theme_provider_1.styled(exports.DemoContent).withConfig({ displayName: "LargeDemoContent", componentId: "sc-dg8mqm" }) `
	height: 150px;
`;
const DemoContainerWrapper = theme_provider_1.styled.div.withConfig({ displayName: "DemoContainerWrapper", componentId: "sc-w2h6ov" }) `
	border: 1px solid ${props => props.theme.text.color.primary};
`;
exports.demoContent = [
    React.createElement(exports.DemoContent, { key: "first_div" }, "First div"),
    React.createElement(exports.DemoContent, { key: "second_div" }, "Second div"),
    React.createElement(exports.DemoContent, { key: "third_div" }, "Third div"),
    React.createElement(exports.DemoContent, { key: "fourth_div" }, "Fourth div"),
    React.createElement(exports.DemoContent, { key: "fifth_div" }, "Fifth div"),
    React.createElement(exports.DemoContent, { key: "sixth_div" }, "Sixth div"),
    React.createElement(exports.DemoContent, { key: "seventh_div" }, "Seventh div"),
    React.createElement(exports.DemoContent, { key: "eighth_div" }, "Eighth div"),
    React.createElement(exports.DemoContent, { key: "ninth_div" }, "Ninth div"),
    React.createElement(exports.DemoContent, { key: "tenth_div" }, "Tenth div")
];
exports.demoContentWithSeparators = [
    React.createElement(exports.DemoContent, { key: "first_div" }, "First div"),
    React.createElement("hr", { key: "first_separator" }),
    React.createElement(exports.DemoContent, { key: "second_div" }, "Second div"),
    React.createElement("hr", { key: "second_separator" }),
    React.createElement(exports.DemoContent, { key: "third_div" }, "Third div"),
    React.createElement("hr", { key: "third_separator" }),
    React.createElement(exports.DemoContent, { key: "fourth_div" }, "Fourth div"),
    React.createElement("hr", { key: "fourth_separator" }),
    React.createElement(exports.DemoContent, { key: "fifth_div" }, "Fifth div"),
    React.createElement("hr", { key: "fifth_separator" }),
    React.createElement(exports.DemoContent, { key: "sixth_div" }, "Sixth div"),
    React.createElement("hr", { key: "sixth_separator" }),
    React.createElement(exports.DemoContent, { key: "seventh_div" }, "Seventh div"),
    React.createElement("hr", { key: "seventh_separator" }),
    React.createElement(exports.DemoContent, { key: "eighth_div" }, "Eighth div"),
    React.createElement("hr", { key: "eighth_separator" }),
    React.createElement(exports.DemoContent, { key: "ninth_div" }, "Ninth div"),
    React.createElement("hr", { key: "ninth_separator" }),
    React.createElement(exports.DemoContent, { key: "tenth_div" }, "Tenth div")
];
const ContainerDemo = () => (React.createElement(index_1.DemoRoot, { showGridOverlayToggle: true },
    React.createElement("h2", null, "Container"),
    React.createElement("p", null, "Please note: the container is a pattern used for arranging it's child components. The child components used here are supposed to visualize the layout, they are not actual patterns."),
    React.createElement("hr", null),
    React.createElement(demo_1.DemoThemeProviderWithToggle, null,
        React.createElement("h3", null, "basic styles"),
        React.createElement("h4", null, "single child"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(index_2.Container, null,
                React.createElement(exports.DemoContent, { key: "first_div" }, "First div"))),
        React.createElement("h4", null, "without gutter between children"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(index_2.Container, null, exports.demoContent)),
        React.createElement("hr", null),
        React.createElement("h3", null, "gutter widths between children"),
        React.createElement("h4", null, "with gutter between children (static100)"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static100 }, exports.demoContent)),
        React.createElement("h4", null, "with gutter between children (static200)"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static200 }, exports.demoContent)),
        React.createElement("h4", null, "with gutter between children (static300)"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static300 }, exports.demoContent)),
        React.createElement("hr", null),
        React.createElement("h3", null, "wrapping modes"),
        React.createElement("h4", null, "with space between children (static100) and wrap 'never'"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static100, wrap: index_2.ContainerWrap.never }, exports.demoContent)),
        React.createElement("h4", null, "with space between children (static100) and wrap 'always'"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static100, wrap: index_2.ContainerWrap.always }, exports.demoContent)),
        React.createElement("h4", null, "with wrap mode 'always' and 'never' on various breakpoints"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static100, wrap: {
                        [helpers_1.Breakpoints.default]: index_2.ContainerWrap.always,
                        [helpers_1.Breakpoints.b960]: index_2.ContainerWrap.never
                    } }, exports.demoContent))),
        React.createElement("h4", null, "with wrap mode 'always' and 'auto' on various breakpoints"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static100, wrap: {
                        [helpers_1.Breakpoints.default]: index_2.ContainerWrap.always,
                        [helpers_1.Breakpoints.b560]: index_2.ContainerWrap.auto
                    } }, exports.demoContent))),
        React.createElement("hr", null),
        React.createElement("h3", null, "stretch content to width of container"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(index_2.Container, { stretchContent: true }, exports.demoContent)),
        React.createElement("hr", null),
        React.createElement("h3", null, "surrounding padding widths"),
        React.createElement("h4", null, "with padding (static100)"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static100, padding: index_2.ContainerPadding.static100 }, exports.demoContent))),
        React.createElement("h4", null, "with padding (static200)"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static100, padding: index_2.ContainerPadding.static200 }, exports.demoContent))),
        React.createElement("hr", null),
        React.createElement("h4", null, "with padding (static300)"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static100, padding: index_2.ContainerPadding.static300 }, exports.demoContent))),
        React.createElement("h4", null, "with different padding for all sides"),
        React.createElement(DemoContainerWrapper, null,
            React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static100, padding: {
                    top: index_2.ContainerPadding.static100,
                    left: {
                        [helpers_1.Breakpoints.b960]: index_2.ContainerPadding.grid001,
                        [helpers_1.Breakpoints.b1280]: index_2.ContainerPadding.grid002
                    }
                } }, exports.demoContent)),
        React.createElement("hr", null),
        React.createElement("h3", null, "Vertical alignment of children"),
        React.createElement("h4", null, "children vertically centered"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static100, verticalAlign: index_2.ContainerVerticalAlignment.center },
                    React.createElement(exports.DemoContent, null, "small content"),
                    React.createElement(exports.LargeDemoContent, null, "large content")))),
        React.createElement("h4", null, "children aligned to the bottom"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static100, verticalAlign: index_2.ContainerVerticalAlignment.flexEnd },
                    React.createElement(exports.DemoContent, null, "small content"),
                    React.createElement(exports.LargeDemoContent, null, "large content")))),
        React.createElement("hr", null),
        React.createElement("h3", null, "Horizontal alignment of children"),
        React.createElement("h4", null, "centered children"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static100, horizontalAlign: index_2.ContainerHorizontalAlignment.center },
                    React.createElement(exports.DemoContent, null, "small content")))),
        React.createElement("h4", null, "children aligned to the end"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static100, horizontalAlign: index_2.ContainerHorizontalAlignment.flexEnd },
                    React.createElement(exports.DemoContent, null, "small content")))),
        React.createElement("h4", null, "children aligned differently on various breakpoints"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static100, horizontalAlign: {
                        [helpers_1.Breakpoints.default]: index_2.ContainerHorizontalAlignment.flexStart,
                        [helpers_1.Breakpoints.b560]: index_2.ContainerHorizontalAlignment.flexEnd,
                        [helpers_1.Breakpoints.b960]: index_2.ContainerHorizontalAlignment.flexStart,
                        [helpers_1.Breakpoints.b1280]: index_2.ContainerHorizontalAlignment.center
                    } },
                    React.createElement(exports.DemoContent, null, "small content")))),
        React.createElement("hr", null),
        React.createElement("h3", null, "Separators between children"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static100, wrap: index_2.ContainerWrap.always }, exports.demoContentWithSeparators))),
        React.createElement("hr", null),
        React.createElement("h3", null, "Container Configuration Options"),
        React.createElement("h4", null, "Container Gutter Configuration"),
        React.createElement("p", null, "using static100, static200, and static300 on various breakpoints"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { gutter: {
                        [helpers_1.Breakpoints.default]: index_2.ContainerGutter.static100,
                        [helpers_1.Breakpoints.tablet]: index_2.ContainerGutter.static200,
                        [helpers_1.Breakpoints.desktop]: index_2.ContainerGutter.static300
                    } }, exports.demoContent))),
        React.createElement("h4", null, "Container Padding Configuration"),
        React.createElement("p", null, "using static100, static200, and static300 on various breakpoints"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { padding: {
                        [helpers_1.Breakpoints.default]: index_2.ContainerPadding.static100,
                        [helpers_1.Breakpoints.tablet]: index_2.ContainerPadding.static200,
                        [helpers_1.Breakpoints.desktop]: index_2.ContainerPadding.static300
                    } }, exports.demoContent))),
        React.createElement("h4", null, "Container Padding Configuration"),
        React.createElement("p", null, "using grid001, grid002, and grid003 on various breakpoints"),
        React.createElement(DemoContainerWrapper, null,
            React.createElement(index_2.Container, { padding: {
                    [helpers_1.Breakpoints.default]: index_2.ContainerPadding.grid001,
                    [helpers_1.Breakpoints.tablet]: index_2.ContainerPadding.grid002,
                    [helpers_1.Breakpoints.desktop]: index_2.ContainerPadding.grid003
                } }, exports.demoContent)),
        React.createElement("h4", null, "Directional Container Padding Configuration"),
        React.createElement("p", null, "using static100, static200, static300, and static400 for different directions on various breakpoints"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { padding: {
                        top: {
                            [helpers_1.Breakpoints.default]: index_2.ContainerPadding.static100,
                            [helpers_1.Breakpoints.tablet]: index_2.ContainerPadding.static200,
                            [helpers_1.Breakpoints.desktop]: index_2.ContainerPadding.static300
                        },
                        bottom: {
                            [helpers_1.Breakpoints.default]: index_2.ContainerPadding.static300,
                            [helpers_1.Breakpoints.tablet]: index_2.ContainerPadding.static400,
                            [helpers_1.Breakpoints.desktop]: index_2.ContainerPadding.static500
                        }
                    } }, exports.demoContent))),
        React.createElement("h4", null, "Directional Container Padding Configuration with LTR / RTL"),
        React.createElement("p", null, "using different sized paddings (static100, static200, static300, and static400) for inlineStart and inlineEnd directions on various breakpoints"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { padding: {
                        inlineStart: {
                            [helpers_1.Breakpoints.default]: index_2.ContainerPadding.static100,
                            [helpers_1.Breakpoints.tablet]: index_2.ContainerPadding.static200,
                            [helpers_1.Breakpoints.desktop]: index_2.ContainerPadding.static300
                        },
                        inlineEnd: {
                            [helpers_1.Breakpoints.default]: index_2.ContainerPadding.static300,
                            [helpers_1.Breakpoints.tablet]: index_2.ContainerPadding.static400,
                            [helpers_1.Breakpoints.desktop]: index_2.ContainerPadding.static500
                        }
                    } }, exports.demoContent))),
        ' ',
        React.createElement("h4", null, "With children that render `null`"),
        React.createElement("p", null, "using static300 as gutter. If any child returns `null` in its render method, no gutter will be reserved"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { gutter: index_2.ContainerGutter.static300 },
                    React.createElement(breakpoint_wrapper_1.BreakpointWrapper, { max: helpers_1.Breakpoints.b960 },
                        React.createElement(exports.DemoContent, null, "I'm only visible below 960px, so there is no gutter for me in larger breakpoints")),
                    exports.demoContent))),
        React.createElement("h4", null, "With reference"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(DemoContainerWrapper, null,
                React.createElement(index_2.Container, { ref: React.createRef() }, exports.demoContent))))));
exports.default = ContainerDemo;
