// @volkswagen-onehub/components-core/dist-es/action-button/index.js
import * as React from 'react';
import { css } from 'styled-components';
import { createA11yOutlineStyles } from '../helpers';
import { styled } from '../theme-provider';
const getColor = (props) => {
    const textRecipe = props.theme.interaction.recipe450;
    return props.disabled
        ? css `
				color: ${textRecipe.disabled};
		  `
        : css `
				& {
					color: ${textRecipe.default};
				}
				:hover,
				:focus {
					color: ${textRecipe.hover};
				}
				:active {
					color: ${textRecipe.active};
				}
		  `;
};
const getBorder = (props) => {
    const borderRecipe = props.theme.interaction.recipe450;
    return props.disabled
        ? css `
				border-color: ${borderRecipe.disabled};
		  `
        : css `
				& {
					border-color: ${borderRecipe.default};
				}
				:hover,
				:focus {
					border-color: ${borderRecipe.hover};
				}
				:active {
					border-color: ${borderRecipe.active};
				}
		  `;
};
const actionButtonStyle = (props) => css `
	min-width: 30px;
	min-height: 30px;
	margin: 0;
	padding: 0 ${props.theme.size.static150};
	border: 1px solid;
	border-radius: 15px;
	outline: none;
	transition: all 0.2s ease-in-out;

	display: inline-flex;
	align-items: center;

	cursor: ${props.disabled ? 'not-allowed' : 'pointer'};
	background: ${props.theme.transparentColor};

	${getColor(props)}
	${getBorder(props)}

	:focus {
		${createA11yOutlineStyles(props)}
	}
`;
const StyledActionButton = styled.button.withConfig({ displayName: "StyledActionButton", componentId: "sc-1drw0xj" }) `
	${actionButtonStyle};
`;
const StyledActionButtonLink = styled.a.withConfig({ displayName: "StyledActionButtonLink", componentId: "sc-t878c9" }) `
	${actionButtonStyle};

	text-decoration: none;
`;
const StyledTextWrap = styled.div.withConfig({ displayName: "StyledTextWrap", componentId: "sc-2ebcn5" }) `
	margin: 0 ${props => props.theme.size.static100};
	display: ${props => (props.reduced ? 'none' : '')};

	// NOTE: this overlaps "margin hack" on Text element due wrong font baseline to get better alignment here
	font-size: ${props => props.theme.textAppearances.copy0100.fontSize};
	position: relative;
	top: 0.11em; // same relative value as in Text component
`;
export const ActionButton = props => {
    const { children, icon, reduced, disabled, onClick, innerRef } = props;
    return (React.createElement(StyledActionButton, { disabled: disabled, onClick: onClick, ref: innerRef },
        icon,
        React.createElement(StyledTextWrap, { reduced: reduced }, children)));
};
export const ActionButtonLink = props => {
    const { children, icon, reduced, disabled, onClick, innerRef, download, target, title, href, rel } = props;
    return (React.createElement(StyledActionButtonLink, { disabled: disabled, onClick: onClick, ref: innerRef, download: download, target: target, title: title, href: href, rel: rel },
        icon,
        React.createElement(StyledTextWrap, { reduced: reduced }, children)));
};
