// @volkswagen-onehub/components-core/dist/top-bar/demo.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const icons_core_1 = require("@volkswagen-onehub/icons-core");
const container_1 = require("../container");
const demo_1 = require("../demo");
const demo_container_1 = require("../demo/demo-container");
const link_1 = require("../link");
const menu_button_1 = require("../menu-button");
const text_1 = require("../text");
const theme_provider_1 = require("../theme-provider");
const index_1 = require("./index");
const Login = () => (React.createElement(text_1.Text, { appearance: text_1.TextAppearance.copy0150, bold: true },
    React.createElement(link_1.Link, { href: "#" }, "Jonathan Longtestname")));
// Secondary content can be eg. in page navigation
class TopBarSecondaryContent extends React.Component {
    render() {
        return (React.createElement("div", null,
            React.createElement(container_1.Container, { wrap: container_1.ContainerWrap.always, gutter: container_1.ContainerGutter.dynamic0020 },
                React.createElement(text_1.Text, null, "Secondary Content Item 1"),
                React.createElement(text_1.Text, null, "Secondary Content Item 2"),
                React.createElement(text_1.Text, null, "Secondary Content Item 3"),
                React.createElement(text_1.Text, null, "Secondary Content Item 4"))));
    }
}
const TopBarDemoMainContainer = theme_provider_1.styled.div.withConfig({ displayName: "TopBarDemoMainContainer", componentId: "sc-we82kg" }) `
	overflow: hidden;
	border-top: 1px solid black;

	/* remove margins and paddings from demo container, add height to show the "page" below the navigation */
	& > div {
		height: 250px;
		padding: 0;
		margin: 0;
	}
`;
const TopBarDemoStickyWrapper = theme_provider_1.styled.div.withConfig({ displayName: "TopBarDemoStickyWrapper", componentId: "sc-1ym4gqx" }) `
	position: sticky;
	top: 0;
`;
class TopBarDemo extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            hideLogo: false
        };
        this.toggleLogo = this.toggleLogo.bind(this);
    }
    toggleLogo() {
        const { hideLogo } = this.state;
        this.setState({ hideLogo: !hideLogo });
    }
    render() {
        const { hideLogo } = this.state;
        return (React.createElement(demo_1.DemoRoot, { showGridOverlayToggle: true },
            React.createElement(demo_1.DemoThemeProviderWithToggle, null,
                React.createElement("h4", null, "Top bar with main menu button, etc."),
                React.createElement("button", { onClick: this.toggleLogo },
                    hideLogo ? 'show' : 'hide',
                    " logo"),
                React.createElement("p", null, "Hiding / showing the logo will be triggered by scrolling on the website. In this demo please use the button instead."),
                React.createElement(TopBarDemoMainContainer, null,
                    React.createElement(demo_container_1.DemoContainer, { hasBackgroundImage: true },
                        React.createElement(TopBarDemoStickyWrapper, null,
                            React.createElement(index_1.TopBar, { hideLogo: hideLogo },
                                React.createElement(menu_button_1.MenuButton, { label: "Open Menu", key: "left" }),
                                React.createElement(container_1.Container, { wrap: container_1.ContainerWrap.never, gutter: container_1.ContainerGutter.dynamic0100, horizontalAlign: container_1.ContainerHorizontalAlignment.flexEnd, verticalAlign: container_1.ContainerVerticalAlignment.center, key: "right" },
                                    React.createElement(Login, null),
                                    React.createElement(icons_core_1.ShoppingCart, null)))))),
                React.createElement("h4", null, "Top bar with secondary content"),
                React.createElement(TopBarDemoMainContainer, null,
                    React.createElement(demo_container_1.DemoContainer, { hasBackgroundImage: true },
                        React.createElement(TopBarDemoStickyWrapper, null,
                            React.createElement(index_1.TopBar, { hideLogo: true },
                                React.createElement(TopBarSecondaryContent, null))))))));
    }
}
exports.default = TopBarDemo;
