// @volkswagen-onehub/components-core/dist/accordion/demo.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const container_1 = require("../container");
const demo_1 = require("../demo");
const layout_1 = require("../layout");
const text_1 = require("../text");
const index_1 = require("./index");
class DemoAccordionStateHolder extends React.PureComponent {
    constructor(props) {
        super(props);
        this.onClick = () => {
            this.setState(state => ({ isOpen: !state.isOpen }));
        };
        this.state = {
            isOpen: props.defaultIsOpen || false
        };
    }
    render() {
        return this.props.children(this.state.isOpen, this.onClick);
    }
}
const sampleHeadline = (React.createElement(text_1.Text, { tag: text_1.TextTag.h2, appearance: text_1.TextAppearance.headline300, bold: true },
    React.createElement(demo_1.DemoPlaceholderText, null)));
const sampleBody = (React.createElement(text_1.Text, { tag: text_1.TextTag.p },
    React.createElement(demo_1.DemoPlaceholderText, null)));
const AccordionDemo = () => (React.createElement(demo_1.DemoRoot, { showGridOverlayToggle: true },
    React.createElement(demo_1.DemoThemeProviderWithToggle, null,
        React.createElement("h2", null, "AccordionDemo"),
        React.createElement(demo_1.DemoContainer, null,
            React.createElement(layout_1.Layout, null,
                React.createElement(container_1.Container, { padding: {
                        left: container_1.ContainerPadding.grid001,
                        right: container_1.ContainerPadding.grid001
                    }, wrap: container_1.ContainerWrap.always },
                    React.createElement(DemoAccordionStateHolder, { defaultIsOpen: true }, (isOpen, onClick) => (React.createElement(index_1.Accordion, { id: "uniqueid0", headline: sampleHeadline, isOpen: isOpen, onClick: onClick }, sampleBody))),
                    React.createElement(DemoAccordionStateHolder, null, (isOpen, onClick) => (React.createElement(index_1.Accordion, { id: "uniqueid1", headline: sampleHeadline, isOpen: isOpen, onClick: onClick }, sampleBody))),
                    React.createElement(DemoAccordionStateHolder, { defaultIsOpen: true }, (isOpen, onClick) => (React.createElement(index_1.Accordion, { id: "uniqueid2", headline: sampleHeadline, isOpen: isOpen, onClick: onClick }, sampleBody))),
                    React.createElement(DemoAccordionStateHolder, null, (isOpen, onClick) => (React.createElement(index_1.Accordion, { id: "uniqueid3", headline: sampleHeadline, isOpen: isOpen, onClick: onClick }, sampleBody)))))))));
exports.default = AccordionDemo;
