// @volkswagen-onehub/components-core/dist-es/tabs/demo.js
import * as React from 'react';
import { DemoContainer } from '../demo/demo-container';
import { DemoRoot } from '../demo/demo-root';
import { DemoThemeProviderWithToggle } from '../demo/demo-theme-provider-with-toggle';
import { Text } from '../text';
import { Tabs } from './index';
export default class TabsDemo extends React.Component {
    render() {
        return (React.createElement(DemoRoot, null,
            React.createElement(DemoThemeProviderWithToggle, null,
                React.createElement(DemoContainer, null,
                    React.createElement(Tabs, { idPrefix: "bla", onChange: (index) => {
                            console.log('selected tab', index);
                        } },
                        { title: 'Title 1', content: React.createElement("div", null, "Child1"), key: 'a' },
                        {
                            title: React.createElement(Text, null, "Title 2"),
                            content: React.createElement("div", null, "Child2"),
                            key: 'b'
                        },
                        { title: 'Title 3', content: React.createElement("div", null, "Child3"), key: 'c' })))));
    }
}
