// @volkswagen-onehub/components-core/dist-es/layer-wrapper/demo.js
import * as React from 'react';
import { Container, ContainerGutter, ContainerHorizontalAlignment, ContainerPadding, ContainerVerticalAlignment, ContainerWrap } from '../container';
import { DemoContainer } from '../demo/demo-container';
import { DemoRoot } from '../demo/demo-root';
import { DemoThemeProviderWithToggle } from '../demo/demo-theme-provider-with-toggle';
import { Text, TextAppearance, TextTag } from '../text';
import { styled } from '../theme-provider';
import { LayerWrapper } from './index';
const DemoContent = () => (React.createElement(Container, { gutter: ContainerGutter.static100, horizontalAlign: ContainerHorizontalAlignment.center, padding: ContainerPadding.static200, verticalAlign: ContainerVerticalAlignment.center, wrap: ContainerWrap.always },
    React.createElement(Text, { tag: TextTag.p, appearance: TextAppearance.copy200 }, "Lorem ipsum dolor sit amet.")));
const StyledDemoContentWrapper1 = styled.div.withConfig({ displayName: "StyledDemoContentWrapper1", componentId: "sc-1kxwb0q" }) `
	position: absolute;
	top: ${props => props.theme.size.dynamic0250};
	left: ${props => props.theme.size.dynamic0250};
	width: ${props => props.theme.size.grid004};
	height: ${props => props.theme.size.grid004};
	background: ${props => props.theme.container.background.primary};
	overflow: hidden;
`;
StyledDemoContentWrapper1.displayName = 'StyledDemoContentWrapper1';
const StyledDemoContentWrapper2 = styled.div.withConfig({ displayName: "StyledDemoContentWrapper2", componentId: "sc-5c6uzs" }) `
	width: ${props => props.theme.size.grid004};
	height: ${props => props.theme.size.grid004};
	background: ${props => props.theme.container.background.secondary};
	overflow: hidden;
`;
StyledDemoContentWrapper2.displayName = 'StyledDemoContentWrapper2';
const LayerWrapperDemo = () => (React.createElement(DemoRoot, null,
    React.createElement(DemoThemeProviderWithToggle, null,
        React.createElement(DemoContainer, null,
            React.createElement(LayerWrapper, null,
                React.createElement(StyledDemoContentWrapper1, null,
                    React.createElement(DemoContent, null)),
                React.createElement(StyledDemoContentWrapper2, null,
                    React.createElement(DemoContent, null)))),
        React.createElement(DemoContainer, null,
            React.createElement(LayerWrapper, { contentCentered: true },
                React.createElement(StyledDemoContentWrapper2, null,
                    React.createElement(Container, { gutter: ContainerGutter.static100, horizontalAlign: ContainerHorizontalAlignment.center, padding: ContainerPadding.static200, verticalAlign: ContainerVerticalAlignment.center, wrap: ContainerWrap.always },
                        React.createElement(Text, { tag: TextTag.p, appearance: TextAppearance.copy200 }, "LayerWrapper with content centered"))))))));
export default LayerWrapperDemo;
