// @volkswagen-onehub/components-core/dist-es/shim-layer/demo.js
import * as React from 'react';
import { Button } from '../button';
import { Checkbox } from '../checkbox';
import { Container, ContainerGutter, ContainerPadding, ContainerWrap } from '../container';
import { DemoContainer } from '../demo/demo-container';
import { DemoRoot } from '../demo/demo-root';
import { DemoThemeProviderWithToggle } from '../demo/demo-theme-provider-with-toggle';
import { LayerWrapper } from '../layer-wrapper';
import { Text, TextAppearance, TextTag } from '../text';
import { ShimBackgroundType, ShimLayer } from './index';
import { demoData } from './demo-data';
class ShimLayerDemo extends React.Component {
    constructor() {
        super(...arguments);
        this.state = {
            layerIsVisible: false,
            layerBackgroundType: ShimBackgroundType.OPAQUE
        };
        this.toggleVisibility = () => {
            this.setState({
                layerIsVisible: !this.state.layerIsVisible
            });
        };
        this.toggleLayerTransparency = () => {
            this.setState({
                layerBackgroundType: this.state.layerBackgroundType === ShimBackgroundType.OPAQUE
                    ? ShimBackgroundType.TRANSPARENT
                    : ShimBackgroundType.OPAQUE
            });
        };
    }
    render() {
        return (React.createElement(DemoRoot, { showGridOverlayToggle: true },
            React.createElement(DemoThemeProviderWithToggle, null,
                React.createElement(DemoContainer, null,
                    React.createElement(Container, { gutter: ContainerGutter.dynamic0100, padding: ContainerPadding.grid001, wrap: ContainerWrap.always },
                        React.createElement(Text, { tag: TextTag.div, appearance: TextAppearance.headline350 }, "Select your shim layer options"),
                        React.createElement(Checkbox, { value: "layer-is-transparent", checked: this.state.layerBackgroundType ===
                                ShimBackgroundType.TRANSPARENT, onChange: this.toggleLayerTransparency, label: "transparent shim layer" }),
                        React.createElement(Button, { onClick: this.toggleVisibility }, "Open Shim Layer"),
                        React.createElement(Text, { tag: TextTag.p },
                            "Click on the shim layer that opens to close it.",
                            React.createElement("br", null),
                            "The layer is currently",
                            ' ',
                            React.createElement(Text, { tag: TextTag.span, bold: true },
                                this.state.layerIsVisible ? 'open' : 'closed',
                                "."))),
                    React.createElement(Container, { gutter: ContainerGutter.static100, padding: ContainerPadding.grid001, wrap: ContainerWrap.always },
                        React.createElement(Text, { tag: TextTag.div, appearance: TextAppearance.headline350 }, "This is some dummy page content"),
                        React.createElement(Text, null, demoData)),
                    React.createElement(LayerWrapper, null, this.state.layerIsVisible ? (React.createElement(ShimLayer, { onClick: this.toggleVisibility, backgroundType: this.state.layerBackgroundType })) : (undefined))))));
    }
}
export default ShimLayerDemo;
