// @volkswagen-onehub/components-core/dist/text-fields/text-input/index.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const theme_provider_1 = require("../../theme-provider");
const shared_1 = require("../shared");
const text_field_clean_wrapper_1 = require("../text-field-clean-wrapper");
const text_field_wrapper_1 = require("../text-field-wrapper");
const utils_1 = require("../utils");
// @todo get from theme
const borderSize = '1px';
const StyledInput = theme_provider_1.styled.input.withConfig({ displayName: "StyledInput", componentId: "sc-7giqxg" }) `
	min-height: ${props => props.theme.size.static300};
	width: 100%;
	padding: ${props => props.icon
    ? `${props.theme.size.static150} ${props.theme.size.static400}`
    : `${props.theme.size.static150} 0`};
	border: 0;
	border-bottom: ${({ appearance }) => appearance !== shared_1.TextInputAppearance.ReadOnly && `${borderSize} solid`};
	border-color: ${props => {
    let color = props.theme.interaction.recipe150.default;
    if (props.appearance === shared_1.TextInputAppearance.Error) {
        color = props.theme.interaction.recipe360.default;
    }
    else if (props.appearance === shared_1.TextInputAppearance.Success) {
        color = props.theme.interaction.recipe350.default;
    }
    return color;
}};
	border-radius: 0;
	background: transparent;
	cursor: ${props => props.appearance === shared_1.TextInputAppearance.Disabled ? 'not-allowed' : ''};

	color: ${utils_1.getColor};
	outline: 0;
	font-size: ${props => props.theme.textAppearances.copy0200.fontSize};

	font-family: inherit;
	margin: ${props => props.theme.size.static100} 0;

	&:focus {
		border-color: ${props => {
    if (props.appearance === shared_1.TextInputAppearance.Error ||
        props.appearance === shared_1.TextInputAppearance.Success) {
        return;
    }
    return props.theme.interaction.recipe150.checked;
}};
		outline-style: solid;
		outline-width: 1px;
		outline-color: ${props => props.theme.interaction.recipe100.active};
		outline-offset: 2px;

		~ ${text_field_wrapper_1.FloatingStyledLabelText} {
			transform: translateY(0);
			font-size: ${props => props.theme.textAppearances.copy100.fontSize};
		}
	}

	${props => props.valid &&
    theme_provider_1.css `
			& ~ ${text_field_wrapper_1.FloatingStyledLabelText} {
				transform: translateY(0);
				font-size: ${props.theme.textAppearances.copy100.fontSize};
			}
		`}

	&::-ms-clear {
		display: none;
	}
`;
/* eslint-disable-next-line react/display-name */
class TextInputInner extends React.Component {
    constructor(props) {
        super(props);
        this.state = { valid: false };
        this.validateInputField = this.validateInputField.bind(this);
    }
    validateInputField(event) {
        this.setState({ valid: Boolean(event.currentTarget.value) });
    }
    render() {
        const { additionalFeedback, appearance, autoFocus, clearIconLabel = '', defaultValue, errorText, hasClearButton, icon, innerRef, isClearButtonVisible, isFloating, label = '', maxLength, name, onBlur, onChange, onKeyDown, onClear, onFocus, required = false, type, value, autoComplete } = this.props;
        const { valid } = this.state;
        return (React.createElement(text_field_wrapper_1.TextFieldWrapper, { appearance: appearance, label: label, errorText: errorText, value: value, required: required, icon: icon, additionalFeedback: additionalFeedback, maxLength: maxLength, isFloating: isFloating },
            React.createElement(text_field_clean_wrapper_1.TextFieldCleanWrapper, { onClear: onClear, hasClearButton: hasClearButton, isClearButtonVisible: isClearButtonVisible, clearIconLabel: clearIconLabel, maxLength: maxLength },
                React.createElement(StyledInput, { appearance: appearance, autoFocus: autoFocus, defaultValue: defaultValue, disabled: appearance === shared_1.TextInputAppearance.Disabled, icon: icon, maxLength: maxLength, name: name, onBlur: event => {
                        this.validateInputField(event);
                        if (onBlur) {
                            onBlur(event);
                        }
                    }, onChange: onChange, onFocus: onFocus, onKeyDown: onKeyDown, readOnly: appearance === shared_1.TextInputAppearance.ReadOnly, ref: innerRef, required: required, type: type, valid: valid, value: value, autoComplete: autoComplete }))));
    }
}
exports.TextInput = React.forwardRef((props, ref) => React.createElement(TextInputInner, Object.assign({}, props, { innerRef: ref })));
