// @volkswagen-onehub/components-core/dist-es/text-fields/text-field-clean-wrapper/index.js
import * as React from 'react';
import { styled } from '../../theme-provider';
import { getLabelIconColor } from '../utils';
import { Close } from '@volkswagen-onehub/icons-core';
import { createA11yOutlineStyles } from '../../helpers';
const StyledClearButton = styled.button.withConfig({ displayName: "StyledClearButton", componentId: "sc-1o29182" }) `
	background: transparent;
	padding: 0;
	border: none;
	line-height: 0;
	cursor: pointer;
	color: ${getLabelIconColor};
	border-bottom: 1px solid transparent;
	pointer-events: initial;
	:hover {
		color: ${props => props.theme.interaction.recipe400.hover};
	}
	:focus {
		color: ${props => props.theme.interaction.recipe400.hover};
		${props => createA11yOutlineStyles(props)}
	}
`;
const StyledClearButtonWrapper = styled.div.withConfig({ displayName: "StyledClearButtonWrapper", componentId: "sc-1guav8u" }) `
	position: absolute;
	right: 0;
	left: 0;
	/* text-input margin + text-input padding + CharacterCounter line-height */
	bottom: ${props => {
    const textInputSpacings = parseInt(props.theme.size.static150) +
        parseInt(props.theme.size.static100);
    if (props.maxLength) {
        return `calc(${textInputSpacings}px + ${props.theme.textAppearances.copy100.lineHeight})`;
    }
    return `${textInputSpacings}px`;
}};
	pointer-events: none;
	display: flex;
	flex-direction: row-reverse;
`;
export const TextFieldCleanWrapper = props => {
    const { appearance, children, onClear, clearIconLabel, hasClearButton, isClearButtonVisible, maxLength } = props;
    return (React.createElement(React.Fragment, null,
        children,
        hasClearButton && isClearButtonVisible && (React.createElement(StyledClearButtonWrapper, { maxLength: maxLength },
            React.createElement(StyledClearButton, { appearance: appearance, "aria-label": clearIconLabel, title: clearIconLabel, onClick: onClear, type: "button" },
                React.createElement(Close, { variant: "default" }))))));
};
TextFieldCleanWrapper.displayName = 'TextFieldCleanWrapper';
