// @volkswagen-onehub/components-core/dist-es/focus-layer/demo.js
import * as React from 'react';
import { Button } from '../button';
import { Checkbox } from '../checkbox';
import { Container, ContainerGutter, ContainerPadding, ContainerWrap } from '../container';
import { DemoContainer } from '../demo/demo-container';
import { DemoRoot } from '../demo/demo-root';
import { DemoThemeProviderWithToggle } from '../demo/demo-theme-provider-with-toggle';
import { withFocusLock } from '../hoc';
import { LayerWrapper } from '../layer-wrapper';
import { RadioButton } from '../radio-button';
import { ShimLayer } from '../shim-layer';
import { Text, TextAppearance } from '../text';
import { FocusLayer, FocusLayerSize } from './index';
import { demoData } from './demo-data';
const FocusLayerWithFocusLock = withFocusLock(FocusLayer);
class FocusLayerDemo extends React.Component {
    constructor() {
        super(...arguments);
        this.state = {
            withOverflowingContent: true,
            layerIsVisible: false,
            layerSize: FocusLayerSize.A
        };
        this.toggleVisibility = () => {
            this.setState(state => ({
                layerIsVisible: !state.layerIsVisible
            }));
        };
        this.toggleContent = (_event, props) => {
            this.setState({
                withOverflowingContent: !props.checked
            });
        };
        this.toggleSize = (_event, props) => {
            this.setState({
                layerSize: props.value
            });
        };
    }
    render() {
        const layerContent = this.state.withOverflowingContent
            ? demoData.long
            : demoData.short;
        const { layerIsVisible, layerSize } = this.state;
        return (React.createElement(DemoRoot, { showGridOverlayToggle: true },
            React.createElement(DemoThemeProviderWithToggle, null,
                React.createElement(DemoContainer, null,
                    React.createElement(Container, { gutter: ContainerGutter.dynamic0100, padding: ContainerPadding.grid001, wrap: ContainerWrap.always },
                        React.createElement(Text, { appearance: TextAppearance.headline350 }, "Select your focus layer options"),
                        React.createElement(Checkbox, { value: "long-text", checked: this.state.withOverflowingContent, onChange: this.toggleContent, label: "with overflowing text" }),
                        React.createElement(RadioButton, { name: "size", value: FocusLayerSize.A, checked: this.state.layerSize === FocusLayerSize.A, onChange: this.toggleSize, label: "Size A" }),
                        React.createElement(RadioButton, { name: "size", value: FocusLayerSize.B, checked: this.state.layerSize === FocusLayerSize.B, onChange: this.toggleSize, label: "Size B" }),
                        React.createElement(Button, { onClick: this.toggleVisibility }, "Open Focus Layer")),
                    React.createElement(Container, { gutter: ContainerGutter.static100, padding: ContainerPadding.grid001, wrap: ContainerWrap.always },
                        React.createElement(Text, { appearance: TextAppearance.headline350 }, "This is some dummy page content"),
                        React.createElement(Text, null, demoData.long)),
                    React.createElement(LayerWrapper, { contentCentered: true }, layerIsVisible
                        ? [
                            React.createElement(ShimLayer, { key: "shim" }),
                            React.createElement(FocusLayerWithFocusLock, { key: "focus-layer", active: layerIsVisible, size: layerSize, ariaLabelledBy: "headline-id", onEscKeyUp: this.toggleVisibility },
                                React.createElement(Container, { gutter: ContainerGutter.dynamic0100, padding: ContainerPadding.grid001, wrap: ContainerWrap.always },
                                    React.createElement("div", { id: "headline-id" },
                                        React.createElement(Text, { bold: true, appearance: TextAppearance.headline0300 }, "Focus layer demo")),
                                    React.createElement(Button, { onClick: this.toggleVisibility }, "Close Focus Layer"),
                                    React.createElement(Container, { gutter: ContainerGutter.dynamic0200 }, layerContent
                                        .split('\n\n')
                                        .map((paragraph, index) => (React.createElement(Text, { key: `paragraph-${index}` }, paragraph)))),
                                    React.createElement(Button, { onClick: this.toggleVisibility }, "Close Focus Layer")))
                        ]
                        : undefined)))));
    }
}
export default FocusLayerDemo;
