// @volkswagen-onehub/components-core/dist-es/richtext/index.js
import Markdown from 'markdown-to-jsx';
import * as React from 'react';
import { Link } from '../link';
import { List, ListItem } from '../list';
import { Text, TextAppearance, TextTag } from '../text';
const { headline0400, headline0350, headline0250, headline0200, copy0200 } = TextAppearance;
const { div, h1, h2, h3, h4, strong, p } = TextTag;
export function createTagOverride(component, props) {
    return { component, props };
}
const defaultMdOptions = {
    overrides: {
        h1: createTagOverride(Text, { tag: h1, appearance: headline0400 }),
        h2: createTagOverride(Text, { tag: h2, appearance: headline0350 }),
        h3: createTagOverride(Text, { tag: h3, appearance: headline0250 }),
        h4: createTagOverride(Text, { tag: h4, appearance: headline0200 }),
        p: createTagOverride(Text, { tag: p, appearance: copy0200 }),
        span: createTagOverride(Text, { appearance: copy0200 }),
        div: createTagOverride(Text, { tag: div, appearance: copy0200 }),
        strong: createTagOverride(Text, { tag: strong, bold: true }),
        a: createTagOverride(Link, { isInline: true }),
        ul: createTagOverride(List),
        ol: createTagOverride(List, { isOrdered: true }),
        li: createTagOverride(ListItem)
    }
};
export const Richtext = props => {
    const { markdown, overrides } = props;
    const mdOptions = {
        overrides: Object.assign({}, defaultMdOptions.overrides, overrides)
    };
    return React.createElement(Markdown, { options: mdOptions }, markdown);
};
