// @volkswagen-onehub/components-core/dist/tabs/demo.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const demo_container_1 = require("../demo/demo-container");
const demo_root_1 = require("../demo/demo-root");
const demo_theme_provider_with_toggle_1 = require("../demo/demo-theme-provider-with-toggle");
const text_1 = require("../text");
const index_1 = require("./index");
class TabsDemo extends React.Component {
    render() {
        return (React.createElement(demo_root_1.DemoRoot, null,
            React.createElement(demo_theme_provider_with_toggle_1.DemoThemeProviderWithToggle, null,
                React.createElement(demo_container_1.DemoContainer, null,
                    React.createElement(index_1.Tabs, { idPrefix: "bla", onChange: (index) => {
                            console.log('selected tab', index);
                        } },
                        { title: 'Title 1', content: React.createElement("div", null, "Child1"), key: 'a' },
                        {
                            title: React.createElement(text_1.Text, null, "Title 2"),
                            content: React.createElement("div", null, "Child2"),
                            key: 'b'
                        },
                        { title: 'Title 3', content: React.createElement("div", null, "Child3"), key: 'c' })))));
    }
}
exports.default = TabsDemo;
