// @volkswagen-onehub/components-core/dist-es/checkbox/index.js
import * as React from 'react';
import { CheckboxBorder, CheckboxCheckmark } from '@volkswagen-onehub/icons-core';
import { Text, TextColor, TextTag } from '../text';
import { styled } from '../theme-provider';
export var CheckboxAppearance;
(function (CheckboxAppearance) {
    CheckboxAppearance[CheckboxAppearance["Error"] = 0] = "Error";
})(CheckboxAppearance || (CheckboxAppearance = {}));
const noop = () => undefined;
const StyledLabel = styled.label.withConfig({ displayName: "StyledLabel", componentId: "sc-tap8jg" }) `
	position: relative;
	color: ${props => props.theme.interaction.recipe450.default};
	cursor: pointer;
`;
const StyledCheckboxWrapper = styled.span.withConfig({ displayName: "StyledCheckboxWrapper", componentId: "sc-1gajbzj" }) `
	position: relative;
	height: ${props => props.theme.textAppearances.copy0200.lineHeight};
	display: flex;
	align-items: center;
`;
const StyledCheckboxCheckmark = styled(CheckboxCheckmark).withConfig({ displayName: "StyledCheckboxCheckmark", componentId: "sc-nf8hca" }) `
	position: absolute;
	opacity: 0;
	display: flex;
	align-items: center;
`;
const StyledFocusWrapper = styled.span.withConfig({ displayName: "StyledFocusWrapper", componentId: "sc-1oeagwv" }) `
	display: grid;
	grid-column-gap: ${props => props.theme.size.static200};
	grid-template-columns: max-content auto;
	/* we need a higher specificity here than :checked or :focus */
	&&&:hover {
		color: ${props => props.theme.interaction.recipe450.hover};
	}
`;
const StyledErrorText = styled.span.withConfig({ displayName: "StyledErrorText", componentId: "sc-uiyaus" }) `
	display: block;
	color: ${props => props.theme.interaction.recipe360.default};
	font-size: ${props => props.theme.textAppearances.copy0100.fontSize};
`;
const StyledInput = styled.input.withConfig({ displayName: "StyledInput", componentId: "sc-h9lzej" }) `
	position: absolute;
	opacity: 0;
	outline: 0;
	user-select: none;
	pointer-events: none;

	:checked {
		+ ${StyledFocusWrapper} {
			color: ${props => props.theme.interaction.recipe450.checked};

			${StyledCheckboxWrapper} {
				${StyledCheckboxCheckmark} {
					opacity: 1;
				}
			}
		}
	}

	:focus {
		+ ${StyledFocusWrapper} {
			outline: 1px solid ${props => props.theme.interaction.recipe100.active};
			outline-offset: 2px;
			color: ${props => props.theme.interaction.recipe450.hover};
		}
	}

	/* we need a higher specificity here than :hover */
	&&&:disabled {
		+ ${StyledFocusWrapper} {
			color: ${props => props.theme.interaction.recipe450.disabled};
			cursor: not-allowed;
		}
	}
`;
export class Checkbox extends React.Component {
    constructor() {
        super(...arguments);
        this.handleChange = (e) => {
            const { onChange = noop } = this.props;
            onChange(e, this.props);
        };
        this.handleBlur = (e) => {
            const { onBlur = noop } = this.props;
            onBlur(e, this.props);
        };
        this.handleFocus = (e) => {
            const { onFocus = noop } = this.props;
            onFocus(e, this.props);
        };
        this.handleClick = (e) => {
            const { onClick = noop } = this.props;
            e.currentTarget.focus();
            onClick(e, this.props);
        };
    }
    render() {
        const { checked, defaultChecked, disabled, label = '', name, value, required, appearance, errorText, tabIndex } = this.props;
        const errorShouldBeVisible = Boolean(appearance === CheckboxAppearance.Error && errorText);
        return (React.createElement(StyledLabel, null,
            React.createElement(StyledInput, { checked: checked, defaultChecked: defaultChecked, disabled: disabled, name: name, onBlur: this.handleBlur, onChange: this.handleChange, onClick: this.handleClick, onFocus: this.handleFocus, type: "checkbox", value: value, required: required, tabIndex: tabIndex }),
            React.createElement(StyledFocusWrapper, null,
                React.createElement(StyledCheckboxWrapper, null,
                    React.createElement(CheckboxBorder, { variant: "default" }),
                    React.createElement(StyledCheckboxCheckmark, { variant: "default" })),
                React.createElement("span", null,
                    label &&
                        (typeof label === 'string' ? (React.createElement(Text, { tag: TextTag.span, color: TextColor.inherit }, label)) : (label)),
                    React.createElement(StyledErrorText, { "aria-live": "assertive", "aria-relevant": "all" }, errorShouldBeVisible && errorText)))));
    }
}
