// @volkswagen-onehub/components-core/dist-es/accordion/demo.js
import * as React from 'react';
import { Container, ContainerPadding, ContainerWrap } from '../container';
import { DemoContainer, DemoRoot, DemoPlaceholderText, DemoThemeProviderWithToggle } from '../demo';
import { Layout } from '../layout';
import { Text, TextAppearance, TextTag } from '../text';
import { Accordion } from './index';
class DemoAccordionStateHolder extends React.PureComponent {
    constructor(props) {
        super(props);
        this.onClick = () => {
            this.setState(state => ({ isOpen: !state.isOpen }));
        };
        this.state = {
            isOpen: props.defaultIsOpen || false
        };
    }
    render() {
        return this.props.children(this.state.isOpen, this.onClick);
    }
}
const sampleHeadline = (React.createElement(Text, { tag: TextTag.h2, appearance: TextAppearance.headline300, bold: true },
    React.createElement(DemoPlaceholderText, null)));
const sampleBody = (React.createElement(Text, { tag: TextTag.p },
    React.createElement(DemoPlaceholderText, null)));
const AccordionDemo = () => (React.createElement(DemoRoot, { showGridOverlayToggle: true },
    React.createElement(DemoThemeProviderWithToggle, null,
        React.createElement("h2", null, "AccordionDemo"),
        React.createElement(DemoContainer, null,
            React.createElement(Layout, null,
                React.createElement(Container, { padding: {
                        left: ContainerPadding.grid001,
                        right: ContainerPadding.grid001
                    }, wrap: ContainerWrap.always },
                    React.createElement(DemoAccordionStateHolder, { defaultIsOpen: true }, (isOpen, onClick) => (React.createElement(Accordion, { id: "uniqueid0", headline: sampleHeadline, isOpen: isOpen, onClick: onClick }, sampleBody))),
                    React.createElement(DemoAccordionStateHolder, null, (isOpen, onClick) => (React.createElement(Accordion, { id: "uniqueid1", headline: sampleHeadline, isOpen: isOpen, onClick: onClick }, sampleBody))),
                    React.createElement(DemoAccordionStateHolder, { defaultIsOpen: true }, (isOpen, onClick) => (React.createElement(Accordion, { id: "uniqueid2", headline: sampleHeadline, isOpen: isOpen, onClick: onClick }, sampleBody))),
                    React.createElement(DemoAccordionStateHolder, null, (isOpen, onClick) => (React.createElement(Accordion, { id: "uniqueid3", headline: sampleHeadline, isOpen: isOpen, onClick: onClick }, sampleBody)))))))));
export default AccordionDemo;
