// @volkswagen-onehub/components-core/dist/layer-wrapper/demo.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const container_1 = require("../container");
const demo_container_1 = require("../demo/demo-container");
const demo_root_1 = require("../demo/demo-root");
const demo_theme_provider_with_toggle_1 = require("../demo/demo-theme-provider-with-toggle");
const text_1 = require("../text");
const theme_provider_1 = require("../theme-provider");
const index_1 = require("./index");
const DemoContent = () => (React.createElement(container_1.Container, { gutter: container_1.ContainerGutter.static100, horizontalAlign: container_1.ContainerHorizontalAlignment.center, padding: container_1.ContainerPadding.static200, verticalAlign: container_1.ContainerVerticalAlignment.center, wrap: container_1.ContainerWrap.always },
    React.createElement(text_1.Text, { tag: text_1.TextTag.p, appearance: text_1.TextAppearance.copy200 }, "Lorem ipsum dolor sit amet.")));
const StyledDemoContentWrapper1 = theme_provider_1.styled.div.withConfig({ displayName: "StyledDemoContentWrapper1", componentId: "sc-1kxwb0q" }) `
	position: absolute;
	top: ${props => props.theme.size.dynamic0250};
	left: ${props => props.theme.size.dynamic0250};
	width: ${props => props.theme.size.grid004};
	height: ${props => props.theme.size.grid004};
	background: ${props => props.theme.container.background.primary};
	overflow: hidden;
`;
StyledDemoContentWrapper1.displayName = 'StyledDemoContentWrapper1';
const StyledDemoContentWrapper2 = theme_provider_1.styled.div.withConfig({ displayName: "StyledDemoContentWrapper2", componentId: "sc-5c6uzs" }) `
	width: ${props => props.theme.size.grid004};
	height: ${props => props.theme.size.grid004};
	background: ${props => props.theme.container.background.secondary};
	overflow: hidden;
`;
StyledDemoContentWrapper2.displayName = 'StyledDemoContentWrapper2';
const LayerWrapperDemo = () => (React.createElement(demo_root_1.DemoRoot, null,
    React.createElement(demo_theme_provider_with_toggle_1.DemoThemeProviderWithToggle, null,
        React.createElement(demo_container_1.DemoContainer, null,
            React.createElement(index_1.LayerWrapper, null,
                React.createElement(StyledDemoContentWrapper1, null,
                    React.createElement(DemoContent, null)),
                React.createElement(StyledDemoContentWrapper2, null,
                    React.createElement(DemoContent, null)))),
        React.createElement(demo_container_1.DemoContainer, null,
            React.createElement(index_1.LayerWrapper, { contentCentered: true },
                React.createElement(StyledDemoContentWrapper2, null,
                    React.createElement(container_1.Container, { gutter: container_1.ContainerGutter.static100, horizontalAlign: container_1.ContainerHorizontalAlignment.center, padding: container_1.ContainerPadding.static200, verticalAlign: container_1.ContainerVerticalAlignment.center, wrap: container_1.ContainerWrap.always },
                        React.createElement(text_1.Text, { tag: text_1.TextTag.p, appearance: text_1.TextAppearance.copy200 }, "LayerWrapper with content centered"))))))));
exports.default = LayerWrapperDemo;
