// @volkswagen-onehub/components-core/dist/layer-wrapper/index.js
"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const react_transition_group_1 = require("react-transition-group");
const theme_provider_1 = require("../theme-provider");
const StyledLayerWrapper = theme_provider_1.styled.div.withConfig({ displayName: "StyledLayerWrapper", componentId: "sc-1bl8kxv" }) `
	position: fixed;
	top: 0;
	right: 0;
	bottom: 0;
	left: 0;
	${props => props.contentCentered &&
    `
		display: flex;
		justify-content: center;
		align-items: center;
	`}
	overflow: hidden;

	// The wrapper must always be mounted for animations to work, So we need to be able to click "through" it
	pointer-events: none;

	// All clicks go through the layer wrapper, so enable it for the children (= actual layers) again
	& > * {
		pointer-events: auto;
	}
`;
StyledLayerWrapper.displayName = 'StyledLayerWrapper';
exports.LayerWrapper = props => {
    const { contentCentered } = props, rest = __rest(props, ["contentCentered"]);
    return (React.createElement(StyledLayerWrapper, { contentCentered: contentCentered },
        React.createElement(react_transition_group_1.TransitionGroup, Object.assign({}, rest, { appear: true }))));
};
