// @volkswagen-onehub/components-core/dist-es/breakpoint-wrapper/index.js
import * as React from 'react';
import { Breakpoints } from '../helpers';
export class BreakpointWrapper extends React.Component {
    constructor(props) {
        super(props);
        this.handleMatchMedia = (event) => {
            this.setState({ isVisible: event.matches });
        };
        let { min, max, ssrBreakpoint } = this.props;
        if (min === Breakpoints.default) {
            min = undefined;
        }
        if (max === Breakpoints.default) {
            max = undefined;
        }
        if (ssrBreakpoint === Breakpoints.default) {
            ssrBreakpoint = undefined;
        }
        const mediaQueryMin = min ? ` and (min-width: ${min}px)` : '';
        const mediaQueryMax = max ? ` and (max-width: ${max - 1}px)` : '';
        this.mediaQuery = `all${mediaQueryMin}${mediaQueryMax}`;
        let isVisible = false;
        if (min && max && ssrBreakpoint) {
            isVisible = min <= ssrBreakpoint && max > ssrBreakpoint;
        }
        else if (min && ssrBreakpoint) {
            isVisible = min <= ssrBreakpoint;
        }
        else if (max && ssrBreakpoint) {
            isVisible = max > ssrBreakpoint;
        }
        else if (ssrBreakpoint) {
            isVisible = true;
        }
        else if (!min) {
            isVisible = true;
        }
        this.state = {
            isVisible
        };
    }
    componentDidMount() {
        this.mediaQueryList = window.matchMedia(this.mediaQuery);
        this.setState({ isVisible: this.mediaQueryList.matches });
        if (!this.mediaQueryList.addEventListener) {
            // Fallback for Safari
            this.mediaQueryList.addListener(this.handleMatchMedia);
            return;
        }
        this.mediaQueryList.addEventListener('change', this.handleMatchMedia);
    }
    componentWillUnmount() {
        if (this.mediaQueryList) {
            if (this.mediaQueryList.removeEventListener) {
                this.mediaQueryList.removeEventListener('change', this.handleMatchMedia);
            }
            else {
                // Fallback for Safari
                this.mediaQueryList.removeListener(this.handleMatchMedia);
            }
        }
    }
    render() {
        const { children } = this.props;
        const { isVisible } = this.state;
        return (isVisible && React.createElement(React.Fragment, null, children)) || null;
    }
}
