// @volkswagen-onehub/components-core/dist-es/radio-button/demo.js
import * as React from 'react';
import { DemoContainer } from '../demo/demo-container';
import { DemoRoot } from '../demo/demo-root';
import { DemoThemeProviderWithToggle } from '../demo/demo-theme-provider-with-toggle';
import { RadioButton } from './index';
const label = 'Lorem ipsum dolor sit amet?';
export default class RadioButtonDemo extends React.Component {
    constructor(props) {
        super(props);
        this.handleClick = (event) => {
            console.log('click', event.currentTarget.nodeName);
        };
        this.handleControlledClick = (event, value) => {
            this.setState({
                controlledDemoValue: value
            });
            console.log('click', event.currentTarget.nodeName);
        };
        this.state = {
            controlledDemoValue: 'c1'
        };
    }
    render() {
        return (React.createElement(DemoRoot, null,
            React.createElement("h2", null, "Radio Button"),
            React.createElement(DemoThemeProviderWithToggle, null,
                React.createElement("h4", null, "default"),
                React.createElement(DemoContainer, null,
                    React.createElement(RadioButton, { value: "1", label: label, onClick: this.handleClick })),
                React.createElement("h4", null, "checked"),
                React.createElement(DemoContainer, null,
                    React.createElement(RadioButton, { value: "2", label: label, onClick: this.handleClick, checked: true })),
                React.createElement("h4", null, "disabled"),
                React.createElement(DemoContainer, null,
                    React.createElement(RadioButton, { name: "disabled", value: "d1", label: label, onClick: this.handleClick, disabled: true })),
                React.createElement("h4", null, "disabled and checked"),
                React.createElement(DemoContainer, null,
                    React.createElement(RadioButton, { value: "dc1", label: label, onClick: this.handleClick, checked: true, disabled: true })),
                React.createElement("h4", null, "Controlled component"),
                React.createElement(DemoContainer, null,
                    React.createElement(RadioButton, { value: "c1", label: label, onClick: event => this.handleControlledClick(event, 'c1'), checked: this.state.controlledDemoValue === 'c1' }),
                    React.createElement(RadioButton, { value: "c2", label: label, onClick: event => this.handleControlledClick(event, 'c2'), checked: this.state.controlledDemoValue === 'c2' })),
                React.createElement("h4", null, "Radio group"),
                React.createElement(DemoContainer, null,
                    React.createElement(RadioButton, { value: "c1", label: label, onClick: this.handleClick, name: "arbitraryName" }),
                    React.createElement(RadioButton, { value: "c2", label: label, onClick: this.handleClick, name: "arbitraryName" })),
                React.createElement("h4", null, "Radio without text label"),
                React.createElement(DemoContainer, null,
                    React.createElement(RadioButton, { value: "e1", onClick: this.handleClick })))));
    }
}
