// @volkswagen-onehub/components-core/dist-es/button-next-best-action/demo.js
import * as React from 'react';
import { AirConditioning, Bin, CertifiedRepair } from '@volkswagen-onehub/icons-core';
import { DemoRoot } from '../demo/demo-root';
import { Text, TextAppearance } from '../text';
import { styled } from '../theme-provider';
import { DemoThemeProviderWithToggle, DemoContainer, DemoThemeProviderToggleContext } from '../demo';
import { NextBestActionButton } from './index';
const NBAWrapper = styled.div.withConfig({ displayName: "NBAWrapper", componentId: "sc-1vf1s7" }) `
	padding: 0 2em;
`;
const icons = [
    React.createElement(AirConditioning, { key: "AirConditioning" }),
    React.createElement(Bin, { key: "Bin" }),
    React.createElement(CertifiedRepair, { key: "CertifiedRepair" })
];
export default class NextBestActionButtonDemo extends React.Component {
    constructor(props) {
        super(props);
        this.state = { iconIndex: 0, expanded: false };
        this.toggleIconIndex = this.toggleIconIndex.bind(this);
        this.toggleExpanded = this.toggleExpanded.bind(this);
    }
    toggleIconIndex() {
        const { iconIndex } = this.state;
        this.setState({ iconIndex: (iconIndex + 1) % icons.length });
    }
    toggleExpanded() {
        const { expanded } = this.state;
        this.setState({ expanded: !expanded });
    }
    render() {
        const { iconIndex, expanded } = this.state;
        return (React.createElement(DemoRoot, null,
            React.createElement(DemoThemeProviderWithToggle, null,
                React.createElement(DemoThemeProviderToggleContext.Consumer, null, value => (React.createElement(React.Fragment, null,
                    React.createElement("h2", null, "Next Best Action Button"),
                    React.createElement("p", null, "Please note: the keys used in this demo cause a re-render of the button right when you change the text direction. You probably don't want to do that in your application."),
                    React.createElement(Text, { appearance: TextAppearance.headline250, bold: true }, "Basic Next Best Action Button"),
                    React.createElement(DemoContainer, null,
                        React.createElement(NBAWrapper, null,
                            React.createElement(NextBestActionButton, { action: {
                                    content: 'Next Best Action',
                                    icon: React.createElement(AirConditioning, null),
                                    onClick: () => console.log('Next Best Action')
                                }, key: `button-1-${value.direction}` }))),
                    React.createElement(Text, { appearance: TextAppearance.headline250, bold: true }, "Next Best Action Button with changing icons"),
                    React.createElement("button", { onClick: this.toggleIconIndex }, "change icon"),
                    React.createElement(DemoContainer, null,
                        React.createElement(NBAWrapper, null,
                            React.createElement(NextBestActionButton, { action: {
                                    content: 'Next Best Action',
                                    icon: icons[iconIndex],
                                    onClick: () => console.log('Next Best Action')
                                }, key: `button-2-${value.direction}` }))),
                    React.createElement(Text, { appearance: TextAppearance.headline250, bold: true }, "Next Best Action Button with expanding text"),
                    React.createElement("button", { onClick: this.toggleExpanded }, "expand"),
                    React.createElement(DemoContainer, null,
                        React.createElement(NBAWrapper, null,
                            React.createElement(NextBestActionButton, { action: {
                                    content: 'Next Best Action',
                                    icon: React.createElement(AirConditioning, null),
                                    onClick: () => console.log('Next Best Action')
                                }, key: `button-3-${value.direction}`, expandContent: expanded })))))))));
    }
}
