// @volkswagen-onehub/components-core/dist/radio-button/index.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const icons_core_1 = require("@volkswagen-onehub/icons-core");
const text_1 = require("../text");
const theme_provider_1 = require("../theme-provider");
const noop = () => undefined;
const StyledLabel = theme_provider_1.styled.label.withConfig({ displayName: "StyledLabel", componentId: "sc-12cr2ii" }) `
	position: relative;
	color: ${props => props.theme.interaction.recipe450.default};
	cursor: pointer;
`;
const StyledRadioWrapper = theme_provider_1.styled.span.withConfig({ displayName: "StyledRadioWrapper", componentId: "sc-1n1yaaw" }) `
	position: relative;
	height: ${props => props.theme.textAppearances.copy0200.lineHeight};
	display: flex;
	align-items: center;
`;
const StyledRadioInner = theme_provider_1.styled(icons_core_1.RadioButtonInner).withConfig({ displayName: "StyledRadioInner", componentId: "sc-1ow7h3g" }) `
	position: absolute;
	display: block;
	opacity: 0;
`;
const StyledFocusWrapper = theme_provider_1.styled.span.withConfig({ displayName: "StyledFocusWrapper", componentId: "sc-1nt8p46" }) `
	display: grid;
	grid-column-gap: ${props => props.theme.size.static200};
	grid-template-columns: max-content auto;

	/* we need a higher specificity here than :checked or :focus */
	&&&:hover {
		color: ${props => props.theme.interaction.recipe450.hover};
	}
`;
const StyledInput = theme_provider_1.styled.input.withConfig({ displayName: "StyledInput", componentId: "sc-grykbk" }) `
	position: absolute;
	opacity: 0;
	outline: 0;
	user-select: none;
	pointer-events: none;


	:checked {
		+ ${StyledFocusWrapper} {
			color: ${props => props.theme.interaction.recipe450.checked};

			${StyledRadioWrapper} {
				${StyledRadioInner} {
					opacity: 1;
				}
			}
		}
	}

	:focus {
		+ ${StyledFocusWrapper} {
			outline: 1px solid ${props => props.theme.interaction.recipe100.active};
			outline-offset: 2px;
			color: ${props => props.theme.interaction.recipe450.active};
		}
	}

	/* we need a higher specificity here than :hover */
	&&&:disabled {
		+ ${StyledFocusWrapper} {
			cursor: not-allowed;
			color: ${props => props.theme.interaction.recipe450.disabled};
			}
		}
	}
`;
class RadioButton extends React.Component {
    constructor() {
        super(...arguments);
        this.handleChange = (e) => {
            const { onChange = noop } = this.props;
            onChange(e, this.props);
        };
        this.handleBlur = (e) => {
            const { onBlur = noop } = this.props;
            onBlur(e, this.props);
        };
        this.handleFocus = (e) => {
            const { onFocus = noop } = this.props;
            onFocus(e, this.props);
        };
        this.handleClick = (e) => {
            const { onClick = noop } = this.props;
            onClick(e, this.props);
        };
    }
    render() {
        const { checked, disabled, id, name, value, label = '' } = this.props;
        return (React.createElement(StyledLabel, null,
            React.createElement(StyledInput, { checked: checked, disabled: disabled, id: id, name: name, onBlur: this.handleBlur, onChange: this.handleChange, onClick: this.handleClick, onFocus: this.handleFocus, type: "radio", value: value }),
            React.createElement(StyledFocusWrapper, null,
                React.createElement(StyledRadioWrapper, null,
                    React.createElement(icons_core_1.RadioButtonBorder, { variant: "default" }),
                    React.createElement(StyledRadioInner, { variant: "default" })),
                React.createElement("span", null, label &&
                    (typeof label === 'string' ? (React.createElement(text_1.Text, { tag: text_1.TextTag.span, color: text_1.TextColor.inherit }, label)) : (label))))));
    }
}
exports.RadioButton = RadioButton;
