// @volkswagen-onehub/components-core/dist-es/top-bar/demo.js
import * as React from 'react';
import { ShoppingCart as ShoppingCartIcon } from '@volkswagen-onehub/icons-core';
import { Container, ContainerGutter, ContainerHorizontalAlignment, ContainerVerticalAlignment, ContainerWrap } from '../container';
import { DemoRoot, DemoThemeProviderWithToggle } from '../demo';
import { DemoContainer } from '../demo/demo-container';
import { Link } from '../link';
import { MenuButton } from '../menu-button';
import { Text, TextAppearance } from '../text';
import { styled } from '../theme-provider';
import { TopBar } from './index';
const Login = () => (React.createElement(Text, { appearance: TextAppearance.copy0150, bold: true },
    React.createElement(Link, { href: "#" }, "Jonathan Longtestname")));
// Secondary content can be eg. in page navigation
class TopBarSecondaryContent extends React.Component {
    render() {
        return (React.createElement("div", null,
            React.createElement(Container, { wrap: ContainerWrap.always, gutter: ContainerGutter.dynamic0020 },
                React.createElement(Text, null, "Secondary Content Item 1"),
                React.createElement(Text, null, "Secondary Content Item 2"),
                React.createElement(Text, null, "Secondary Content Item 3"),
                React.createElement(Text, null, "Secondary Content Item 4"))));
    }
}
const TopBarDemoMainContainer = styled.div.withConfig({ displayName: "TopBarDemoMainContainer", componentId: "sc-we82kg" }) `
	overflow: hidden;
	border-top: 1px solid black;

	/* remove margins and paddings from demo container, add height to show the "page" below the navigation */
	& > div {
		height: 250px;
		padding: 0;
		margin: 0;
	}
`;
const TopBarDemoStickyWrapper = styled.div.withConfig({ displayName: "TopBarDemoStickyWrapper", componentId: "sc-1ym4gqx" }) `
	position: sticky;
	top: 0;
`;
class TopBarDemo extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            hideLogo: false
        };
        this.toggleLogo = this.toggleLogo.bind(this);
    }
    toggleLogo() {
        const { hideLogo } = this.state;
        this.setState({ hideLogo: !hideLogo });
    }
    render() {
        const { hideLogo } = this.state;
        return (React.createElement(DemoRoot, { showGridOverlayToggle: true },
            React.createElement(DemoThemeProviderWithToggle, null,
                React.createElement("h4", null, "Top bar with main menu button, etc."),
                React.createElement("button", { onClick: this.toggleLogo },
                    hideLogo ? 'show' : 'hide',
                    " logo"),
                React.createElement("p", null, "Hiding / showing the logo will be triggered by scrolling on the website. In this demo please use the button instead."),
                React.createElement(TopBarDemoMainContainer, null,
                    React.createElement(DemoContainer, { hasBackgroundImage: true },
                        React.createElement(TopBarDemoStickyWrapper, null,
                            React.createElement(TopBar, { hideLogo: hideLogo },
                                React.createElement(MenuButton, { label: "Open Menu", key: "left" }),
                                React.createElement(Container, { wrap: ContainerWrap.never, gutter: ContainerGutter.dynamic0100, horizontalAlign: ContainerHorizontalAlignment.flexEnd, verticalAlign: ContainerVerticalAlignment.center, key: "right" },
                                    React.createElement(Login, null),
                                    React.createElement(ShoppingCartIcon, null)))))),
                React.createElement("h4", null, "Top bar with secondary content"),
                React.createElement(TopBarDemoMainContainer, null,
                    React.createElement(DemoContainer, { hasBackgroundImage: true },
                        React.createElement(TopBarDemoStickyWrapper, null,
                            React.createElement(TopBar, { hideLogo: true },
                                React.createElement(TopBarSecondaryContent, null))))))));
    }
}
export default TopBarDemo;
