// @volkswagen-onehub/components-core/dist-es/navigation-level-list/index.js
import * as React from 'react';
import Fade from 'react-reveal/Fade';
import { css } from 'styled-components';
import { Container, ContainerHorizontalAlignment, ContainerVerticalAlignment, ContainerWrap, ContainerGutter } from '../container';
import { Breakpoints } from '../helpers';
import { Text, TextAppearance, TextColor } from '../text';
import { styled } from '../theme-provider';
import { ChevronDown } from '@volkswagen-onehub/icons-core';
const animationDuration = 300;
const StyledNavigationLevelList = styled.div.withConfig({ displayName: "StyledNavigationLevelList", componentId: "sc-ju1w4n" }) `
	position: relative;
	padding-bottom: ${props => props.theme.size.static150};
`;
const StyledNavigationLevelListItem = styled.div.withConfig({ displayName: "StyledNavigationLevelListItem", componentId: "sc-kdwnmt" }) `
	:not(:first-child) {
		margin-top: ${props => props.theme.size.dynamic0100};
	}

	@media (min-width: ${Breakpoints.b560}px) {
		display: grid;
		grid-column-gap: 1em;
		grid-template-columns: 1fr 1fr;
	}
`;
const StyledIcon = styled(ChevronDown).withConfig({ displayName: "StyledIcon", componentId: "sc-43lik5" }) `
	color: ${props => props.theme.interaction.recipe400.default};
	transition: transform ${animationDuration}ms ease;
	transform: ${props => (props.isOpen ? 'rotateX(180deg)' : 'rotateX(0)')};
	transform-origin: 50% 50%;

	@media (min-width: ${Breakpoints.b560}px) {
		transform: rotateZ(-90deg);
	}
`;
const StyledText = styled.div.withConfig({ displayName: "StyledText", componentId: "sc-1usxyke" }) `
	position: relative;

	::after {
		content: '';
		display: block;
		position: absolute;
		width: 0;
		height: 1px;
		margin-top: ${props => props.theme.size.static100};
		background: ${props => props.theme.interaction.recipe400.checked};
		transition: width ${animationDuration}ms ease;
	}

	${props => props.isOpen &&
    `
			::after {
				width: 100%;
			}
		`};
`;
const LinkStyle = css `
	display: block;
	text-decoration: none;
	color: inherit;
	cursor: pointer;
	-webkit-tap-highlight-color: transparent;

	@media (hover: hover) {
		&:hover {
			color: ${props => props.theme.interaction.recipe100.hover};

			${StyledIcon} {
				color: ${props => props.theme.interaction.recipe100.hover};
			}
		}
	}

	:focus {
		outline: 0;
		color: ${props => props.theme.interaction.recipe100.hover};

		${StyledIcon} {
			color: ${props => props.theme.interaction.recipe100.hover};
		}
	}

	${props => props.isOpen &&
    css `
			@media (hover: hover) {
				color: ${props.theme.interaction.recipe100.checked};

				${StyledIcon} {
					color: ${props.theme.interaction.recipe100.checked};
				}

				&:hover,
				&:focus {
					color: ${props.theme.interaction.recipe100.checked};

					${StyledIcon} {
						color: ${props.theme.interaction.recipe100.checked};
					}
				}
			}
		`};
`;
const StyledLink = styled.a.withConfig({ displayName: "StyledLink", componentId: "sc-1qpvfum" }) `
	${LinkStyle}
`;
const StyledButton = styled.button.withConfig({ displayName: "StyledButton", componentId: "sc-1p7ikll" }) `
	${LinkStyle}

	border: 0;
	padding: 0;
	width: 100%;
	background: transparent;
`;
const StyledSecondLevelList = styled.div.withConfig({ displayName: "StyledSecondLevelList", componentId: "sc-1rmprda" }) `
	${props => props.isOpen &&
    css `
			padding: 0 ${props.theme.size.grid001} ${props.theme.size.dynamic0100};
		`};

	${StyledLink} {
		display: inline-block;
		margin-top: ${props => props.theme.size.dynamic0100};
	}

	@media (min-width: ${Breakpoints.b560}px) {
		position: absolute;
		top: 0;
		padding: 0;

		${StyledLink} {
			:first-child {
				margin-top: 0;
			}
		}
	}
`;
const StyledSecondLevelListInner = styled.div.withConfig({ displayName: "StyledSecondLevelListInner", componentId: "sc-8nshfb" }) `
	display: flex;
	flex-direction: column;
`;
export const NavigationLevelListItem = props => {
    const { id, isOpen = false, name, url, subItems, onClick } = props;
    const firstLevelId = `${id}_firstlevel`;
    const secondLevelId = `${id}_secondlevel`;
    return (React.createElement(StyledNavigationLevelListItem, { id: id },
        React.createElement("div", null, subItems ? (React.createElement(StyledButton, { id: firstLevelId, isOpen: isOpen, onClick: onClick, "aria-expanded": isOpen, "aria-controls": secondLevelId },
            React.createElement(Container, { wrap: ContainerWrap.never, horizontalAlign: ContainerHorizontalAlignment.spaceBetween, verticalAlign: ContainerVerticalAlignment.center, gutter: ContainerGutter.grid001 },
                React.createElement(StyledText, { isOpen: isOpen },
                    React.createElement(Text, { color: TextColor.inherit, appearance: TextAppearance.headline0250 }, name)),
                React.createElement(StyledIcon, { isOpen: isOpen, variant: "small" })))) : (React.createElement(StyledLink, { href: url },
            React.createElement(Text, { color: TextColor.inherit, appearance: TextAppearance.headline0250 }, name)))),
        React.createElement("div", null,
            React.createElement(StyledSecondLevelList, { id: secondLevelId, isOpen: isOpen, "aria-hidden": !isOpen, "aria-labelledby": firstLevelId }, isOpen && (React.createElement(Fade, { duration: animationDuration },
                React.createElement(StyledSecondLevelListInner, null, (subItems || []).map(item => (React.createElement(StyledLink, { href: item.url, key: item.name },
                    React.createElement(Text, { color: TextColor.inherit, appearance: TextAppearance.headline0250 }, item.name)))))))))));
};
NavigationLevelListItem.displayName = 'NavigationLevelListItem';
export const NavigationLevelList = props => (React.createElement(StyledNavigationLevelList, null, props.children));
NavigationLevelList.displayName = 'NavigationLevelList';
