// @volkswagen-onehub/components-core/dist-es/integrator-root/index.js
import { createGlobalStyle } from '@volkswagen-onehub/styled-components-injectglobal-polyfill';
import * as React from 'react';
import { normalize } from 'styled-normalize';
import { default as vhCheck } from 'vh-check';
import { styled, css } from '../theme-provider';
const GlobalStyle = createGlobalStyle `
	${
// eslint-disable-next-line @typescript-eslint/no-explicit-any
normalize};

	b, strong {
		font-weight: bold;
	}
`;
import { isMobileBrowser } from '../helpers';
import { getMediaQueryVars } from '../theme-provider/get-media-query-vars';
// Please do not import this const. It is meant to be used internally only.
export const _fontFamilyStyles = css `
	--font-family-head: vw-head, Helvetica, Arial, sans-serif;
	--font-family-text: vw-text, Helvetica, Arial, sans-serif;

	:lang(ko),
	:lang(zh),
	:lang(tw),
	:lang(ja) {
		font-style: normal;
	}

	:lang(ko) {
		--font-family-head: vw-head, Apple SD Gothic Neo, BonGothic, 맑은 고딕,
			Malgun Gothic, 굴림, Gulim, 돋움, Dotum, Helvetica Neue, Helvetica, Arial,
			sans-serif;
		--font-family-text: vw-text, Apple SD Gothic Neo, BonGothic, 맑은 고딕,
			Malgun Gothic, 굴림, Gulim, 돋움, Dotum, Helvetica Neue, Helvetica, Arial,
			sans-serif;
		word-break: keep-all;
		hyphens: auto; /* Needed for breaking long URLs */
	}

	:lang(zh) {
		--font-family-head: vw-head, 华文细黑, STXihei, PingFang TC, 微软雅黑体,
			Microsoft YaHei New, 微软雅黑, Microsoft Yahei, 宋体, SimSun,
			Helvetica Neue, Helvetica, Arial, sans-serif;
		--font-family-text: vw-text, 华文细黑, STXihei, PingFang TC, 微软雅黑体,
			Microsoft YaHei New, 微软雅黑, Microsoft Yahei, 宋体, SimSun,
			Helvetica Neue, Helvetica, Arial, sans-serif;
	}

	:lang(tw) {
		--font-family-head: vw-head, Helvetica Neue, Helvetica, Arial, sans-serif;
		--font-family-text: vw-text, Helvetica Neue, Helvetica, Arial, sans-serif;
	}

	:lang(ja) {
		--font-family-head: vw-head, ヒラギノ角ゴ Pro W3,
			Hiragino Kaku Gothic Pro, Osaka, メイリオ, Meiryo, MS Pゴシック,
			MS PGothic, Helvetica Neue, Helvetica, Arial, sans-serif;
		--font-family-text: vw-text, ヒラギノ角ゴ Pro W3,
			Hiragino Kaku Gothic Pro, Osaka, メイリオ, Meiryo, MS Pゴシック,
			MS PGothic, Helvetica Neue, Helvetica, Arial, sans-serif;
	}

	font-family: var(--font-family-text);
`;
export const StyledIntegratorRoot = styled.div.withConfig({ displayName: "StyledIntegratorRoot", componentId: "sc-7fx0kf" }) `
	${_fontFamilyStyles}

	-webkit-font-smoothing: antialiased;
	-moz-osx-font-smoothing: grayscale;
	* {
		box-sizing: border-box;
	}
	${getMediaQueryVars()};
`;
const BROWSER_ADDRESS_BAR_CSS_VAR_NAME = 'vwa-components-core-browser-address-bar';
/**
 * On mobile browsers address bar isn't taken into account when using `vh` units. This isn't always desired so this method provides computed value that you can use to calculate proper height.
 *
 * For desktop browsers address bar height is 0px.
 *
 * Example:
 * ```
 * main {
 *   height: 100vh; // falback
 *   height: calc(100vh - ${getBrowserBarHeightValue()});
 * }
 * ```
 */
export function getBrowserBarHeightValue() {
    return `var(--${BROWSER_ADDRESS_BAR_CSS_VAR_NAME}, 0px)`;
}
function updateCssVar(cssVarName, value) {
    document.documentElement.style.setProperty(`--${cssVarName}`, `${value}px`);
}
export class IntegratorRoot extends React.Component {
    constructor() {
        super(...arguments);
        this.updateVh = () => {
            if (this.vhCheckResult) {
                const result = this.vhCheckResult.recompute();
                updateCssVar(BROWSER_ADDRESS_BAR_CSS_VAR_NAME, result.value);
            }
        };
        this.tryUpdateVh = () => {
            if (isMobileBrowser()) {
                this.timeout = window.setTimeout(() => {
                    // on iPhone window height isn't calculated properly at the first try
                    if (window.innerHeight > window.screen.height) {
                        this.tryUpdateVh();
                    }
                    else {
                        window.requestAnimationFrame(this.updateVh);
                    }
                }, 50);
            }
        };
    }
    componentDidMount() {
        this.vhCheckResult = vhCheck(BROWSER_ADDRESS_BAR_CSS_VAR_NAME);
        this.tryUpdateVh();
    }
    componentDidUpdate() {
        this.tryUpdateVh();
    }
    componentWillUnmount() {
        // remove all listeners that are registered by vh-check
        if (this.vhCheckResult) {
            this.vhCheckResult.unbind();
        }
        if (this.timeout) {
            window.clearTimeout(this.timeout);
            this.timeout = undefined;
        }
    }
    render() {
        const { language } = this.props;
        return (React.createElement(StyledIntegratorRoot, { lang: language },
            React.createElement(GlobalStyle, null),
            this.props.children));
    }
}
